docs/developer-setup.md

```diff
@@ -0,0 +1,178 @@
+
+# Development Workflow
+
+## Prerequisites
+
+* You have Go 1.13+ installed on your local development machine.
+* You have Docker installed on your local development machine. Docker is required for building cstor-csi driver container images and to push them into a Kubernetes cluster for testing.
+* You have `kubectl` installed. For running integration tests, you will require an existing single node cluster with [openebs](https://blog.openebs.io/how-to-install-openebs-with-kubernetes-using-minikube-2ed488dff1c2). Don't worry if you don't have access to the Kubernetes cluster, raising a PR with the cstor-csi repository will run integration tests for your changes against a Minikube cluster.
+
+## Initial Setup
+
+### Fork in the cloud
+
+1. Visit https://github.com/openebs/cstor-csi
+2. Click the `Fork` button (top right) to establish a cloud-based fork.
+
+### Clone fork to the local machine
+
+Place openebs/cstor-csi code on your `GOPATH` using the following cloning procedure.
+Create your clone:
+
+```sh
+
+mkdir -p $GOPATH/src/github.com/openebs
+cd $GOPATH/src/github.com/openebs
+
+# Note: Here user= your github profile name
+git clone https://github.com/$user/cstor-csi.git
+
+# Configure remote upstream
+cd $GOPATH/src/github.com/openebs/cstor-csi
+git remote add upstream https://github.com/openebs/cstor-csi.git
+
+# Never push to upstream master
+git remote set-url --push upstream no_push
+
+# Confirm that your remotes make sense:
+git remote -v
+```
+> **Note:** If your `GOPATH` has more than one (`:` separated) paths in it, then you should use *one of your go path* instead of `$GOPATH` in the commands mentioned here. This statement holds throughout this document.
+
+### Building and Testing your changes
+
+* To build the cstor csi-driver binary
+```sh
```

Comment:
nit: is sh required?

Reply:
its just choice how you want to show it, its fine  guess